### PR TITLE
Add button to locate a stop point from the editor form on the map

### PR DESCRIPTION
**Pull request by solita-sabinaf**
Screenshot 2024-11-13 at 11 43 21

### CODE REVIEW

File: src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx [hunk 1]
```diff
@@ -80,11 +92,29 @@ export const GenericStopPointEditor = ({
...
-        />
+        <div style={{ display: 'flex', flexDirection: 'column' }}>
+          <DeleteButton
+            thin={true}
+            disabled={!canDelete}
+            onClick={() => setDeleteDialogOpen(true)}
+            title={formatMessage({ id: 'editorDeleteButtonText' })}
+          />
+
+          {!flexibleLineType && isMapEnabled && (
```
Comment (solita-sabinaf) [inline src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx:103]: !

File: src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx [hunk 2]
```diff
@@ -80,7 +89,29 @@ const JourneyPatternStopPointMap = memo(
...
+    // When "locate stop point" was clicked from GenericStopPointEditor:
+    useEffect(() => {
+      if (focusedQuayId && totalQuayLocationsIndex[focusedQuayId]) {
+        const focusedStopPlaceId: string =
+          totalQuayStopPlaceIndex[focusedQuayId];
+        // Let's produce a proper focusedMarker out of this
+        const newFocusedMarker: FocusedMarker = {
+          stopPlaceId: focusedStopPlaceId,
+          // This array won't play any role in this particular case, so can be just left empty
+          stopPlaceQuays: [],
```
Comment (solita-sabinaf) [inline src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx:103]: Get back to this

File: src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx [hunk 3]
```diff
@@ -20,6 +20,10 @@ import {
 } from '../common/FrontTextTextField';
 import { QuayRefField, useOnQuayRefChange } from '../common/QuayRefField';
 import { StopPointEditorProps } from '../common/StopPointEditorProps';
+import { PositionIcon } from '@entur/icons';
+import '../styles.scss';
```
Comment (testower) [inline src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx:24]: I think this stylesheet is imported elsewhere (I would expect it to be) so it's better to create a new stylesheet file in the same folder, and include it here?

File: src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx [hunk 4]
```diff
@@ -80,11 +91,29 @@ export const GenericStopPointEditor = ({
...
+        <div className={'buttons-group'}>
+          <DeleteButton
+            thin={true}
+            disabled={!canDelete}
+            onClick={() => setDeleteDialogOpen(true)}
+            title={formatMessage({ id: 'editorDeleteButtonText' })}
+          />
+
+          {isLocateButtonAvailable && (
+            <TertiaryButton
```
Comment (testower) [inline src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx:103]: This is an interesting design question. I feel like we are leaking sandbox feature details into the main code here.

File: src/components/StopPointsEditor/Generic/GenericStopPointsEditor.tsx [hunk 5]
```diff
@@ -74,6 +85,8 @@ export const GenericStopPointsEditor = ({
           addStopPoint={addStopPoint}
           deleteStopPoint={deleteStopPoint}
           transportMode={transportMode}
+          focusedQuayId={focusedQuayId}
+          updateFocusedQuayIdCallback={updateFocusedQuayIdCallback}
```
Comment (testower) [inline src/components/StopPointsEditor/Generic/GenericStopPointsEditor.tsx:89]: I think this prop is better named "onFocusedQuayIdUpdate"

File: src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx [hunk 6]
```diff
@@ -29,7 +35,10 @@ const JourneyPatternStopPointMap = memo(
     addStopPoint,
     deleteStopPoint,
     transportMode,
+    focusedQuayId,
+    updateFocusedQuayIdCallback,
   }: JourneyPatternStopPointMapProps) => {
```
Comment (testower) [inline src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx:40]: There's almost 200 lines of business logic before we get to the rendering part here. Can you consolidate some of this aggregate hooks maybe, and separate out into a dedicated file?

Comment (testower) [reply src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx:40]: It also makes it hard to follow the changes.

File: src/ext/JourneyPatternStopPointMap/Popovers/SearchResult.tsx [hunk 7]
```diff
@@ -3,14 +3,13 @@ import { Heading5 } from '@entur/typography';
...
 import { TertiarySquareButton } from '@entur/button';
-import { useMap } from 'react-leaflet';
-import { getStopPlaceLocation } from '../StopPlaceMarker';
 import {
   FocusedMarker,
   JourneyPatternMarker,
   JourneyPatternMarkerType,
 } from '../types';
 import { determineQuayToFocus } from '../helpers';
+import { getStopPlaceLocation } from '../StopPlaceMarker';
```
Comment (testower) [inline src/ext/JourneyPatternStopPointMap/Popovers/SearchResult.tsx:12]: This function seems to belong in a helpers / utility file we can get rid of this awkward dependency to the StopPlaceMarker file.

File: src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx (hunk 4 above)
Comment (solita-sabinaf) [reply src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx:103]: Agreed! I will try to move this into a component and plug in as a Sandbox feature under JourneyPatternStopPointMap folder. But there are a couple more places with similar problem but would be more cumbersome to avoid

Comment (solita-sabinaf) [reply src/components/StopPointsEditor/Generic/GenericStopPointEditor.tsx:103]: Although, this particular case starts to seem bit cumbersome too 😅

File: src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx (hunk 6 above)
Comment (solita-sabinaf) [reply src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx:40]: Yup, I don't like this too. There was only little amount of stuff possible to move into hooks file, so the rest got extracted into a new MarkersWrapper component. Now there will be two components of about 150 lines instead of one gigantic one. Not sure how much it makes things easier to follow in practice, but at least it's not all cramped in one file 😅
